src/WebJobs.Script/Diagnostics/FileLogger.cs

```diff
@@ -27,6 +28,9 @@ public FileLogger(string categoryName, IFileWriter fileWriter, Func<bool> isFile
             _categoryName = categoryName;
             _logType = logType;
             _scopeProvider = scopeProvider ?? throw new ArgumentNullException(nameof(scopeProvider));
+
+            _flushLogs = () => _fileWriter.Flush();
+            _flushLogs = _flushLogs.Debounce();
```

Comment:
Default debounce interval is 300ms which I think is fine here. We can adjust if needed based on load testing which we're doing.